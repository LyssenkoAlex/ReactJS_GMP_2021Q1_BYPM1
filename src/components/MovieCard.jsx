/* eslint-disable jsx-a11y/anchor-is-valid,no-undef,react/prop-types,react/destructuring-assignment,class-methods-use-this,no-param-reassign */
import React from 'react';
import imageNotFound from '../assets/img/image_not_found.jpg';

class MovieCard extends React.Component {
  addDefaultSrc(ev) {
    ev.target.src = imageNotFound;
  }

  render() {
    return (
      <section
        className="movie_container"
        key={`poster_${this.props.movie.id}`}
      >
        <img
          onError={this.addDefaultSrc}
          src={this.props.movie.poster_path}
          alt="headline"
          key={`img_${this.props.movie.id}`}
        />

        <a
          href={`#popup_${this.props.movie.id}`}
          key={`ahref_${this.props.movie.id}`}
        >
          <div className="dots" key={`dots_${this.props.movie.id}`} />
        </a>

        <div
          id={`overlay_${this.props.movie.id}`}
          className="overlay"
          key={`overlay_${this.props.movie.id}`}
        >
          <div className="popup" key={`popup_${this.props.movie.id}`}>
            <a
              className="close"
              href="#"
              key={`a_cl_${this.props.movie.id}`}
            >
              <div key={`times_${this.props.movie.id}`}>&times;</div>
            </a>
            <h5 key={`h5_1_${this.props.movie.id}`}>Edit</h5>
            <h5 key={`h5_2_${this.props.movie.id}`}>Delete</h5>
          </div>
        </div>
      </section>
    );
  }
}

export default MovieCard;
