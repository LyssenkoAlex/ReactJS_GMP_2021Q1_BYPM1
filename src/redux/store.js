import { createStore, applyMiddleware } from "redux";
import rootReducer from "./reducers/reducers";
import apiMiddleware from "./middleware";

const store = createStore(rootReducer, applyMiddleware(apiMiddleware));
window.store = store;
export default store;
