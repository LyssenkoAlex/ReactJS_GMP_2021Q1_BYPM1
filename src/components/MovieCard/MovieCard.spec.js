import React from 'react';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import { BrowserRouter } from 'react-router-dom';
import MovieCard from './MovieCard';
import controlMovie from '../../redux/reducers/_test_/control_movie.json';
import store from '../../redux/store';

describe(`Test MovieCard `, () => {
  it('renders Welcome', () => {
    const tree = renderer
      .create(
        <Provider store={store}>
          <BrowserRouter>
            <MovieCard movie={controlMovie} deleteMovie={() => {}} editMovie={() => {}} />{' '}
          </BrowserRouter>
        </Provider>,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
