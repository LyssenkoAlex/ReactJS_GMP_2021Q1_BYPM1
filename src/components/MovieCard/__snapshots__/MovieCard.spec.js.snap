// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test MovieCard  renders Welcome 1`] = `
Array [
  <section
    className="movie_container"
  >
    <a
      href="/movie_details/337167"
      onClick={[Function]}
    >
      <img
        alt="Fifty Shades Freed"
        onClick={[Function]}
        role="img"
        src="https://image.tmdb.org/t/p/w500/3kcEGnYBHDeqmdYf8ZRbKdfmlUy.jpg"
      />
    </a>
    <section
      className="movies_desc"
    >
      <h3>
        Fifty Shades Freed
      </h3>
      <section
        className="second_line"
      >
        <h3>
          Drama, Romance
        </h3>
        <h4>
          2018
        </h4>
      </section>
    </section>
    <div
      className="dots"
    >
      <div
        className="overlay"
      >
        <div
          className="popup"
        >
          <span
            className="close"
          >
            <div>
              Ã—
            </div>
          </span>
          <h5
            onClick={[Function]}
          >
            Edit
          </h5>
          <h5
            onClick={[Function]}
          >
            Delete
          </h5>
        </div>
      </div>
    </div>
  </section>,
  " ",
]
`;
